client/src/lib.rs

```diff
@@ -279,7 +288,7 @@ pub enum ClientInitError {
 pub struct ConfiguredClient<C> {
     client: C,
     options: Arc<ClientOptions>,
-    headers: Arc<RwLock<HashMap<String, String>>>,
+    headers: Arc<RwLock<ClientHeaders>>,
```

Comment:
I'm working on dynamic endpoints, updating the options field and reconnecting. Is the idea that update headers, that do not need reconnect, with always be updated with `set_headers()` and not by modifying `options`?

Reply:
Yes, that idea has already been the case (`set_headers` has already existed, this just moves things around a bit). I think we need to differentiate things that don't require reconnect (i.e. HTTP headers) from things that do (i.e. hostname and TLS options).